### PR TITLE
zr-1: add email verification

### PR DESCRIPTION
**Pull request by canterbery**
## Describe your changes
 Email verification BE

Added is_verified , verification_code fields to user table
image

Added email module, mailer module , findByEmail method in userService , confirm-email endpoint to auth controller

## Issue ticket code (and/or) and link
- [Link to JIRA ticket](https://testhomework4.atlassian.net/browse/AG-48)

### **General**
- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [ ] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [x] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Backend
- [x] Swagger documentation updated
- [x] Database requests are optimized and not redundant
- [x] Unit tests written
- [x] use ConfigService instead of process.env
- [ ] use transactions if there is a call chain that mutates data in different tables
- [ ] use @index decorator for frequently requested data

### CODE REVIEW

File: src/modules/auth/auth.controller.ts [hunk 1]
```diff
@@ -32,4 +33,13 @@ export class AuthController {
...
     return req.user;
   }
+
+  @Post('confirm-email')
+  @ApiOperation({ summary: 'Confirm email address' })
+  @ApiResponse({ status: 200, description: 'Email address successfully confirmed' })
+  @ApiResponse({ status: 401, description: 'Unauthorized' })
+  async confirmEmail(@Body() confirmEmailDto: ConfirmEmailDto): Promise<UpdateResult> {
+    const { code, email } = confirmEmailDto;
+    return this.authService.confirmEmail(email, code);
```
Comment (lyudad) [inline src/modules/auth/auth.controller.ts]: а кто перехватит ошибку? не вижу трай кеч или врапера. если пропустила можно оставить, если нет обработки - добавить

File: src/modules/email/email.service.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    private userService: UserService,
+    @Inject(forwardRef(() => ConfigService))
+    private config: ConfigService,
+    private mailerService: MailerService,
+  ) {}
+
+  async sendEmailVerificationCode(email: string): Promise<any> {
+    const user = await this.userService.findByEmail(email);
+
+    console.log(user);
```
Comment (lyudad) [inline src/modules/email/email.service.ts]: https://www.notion.so/Git-rules-849f8ca8b3db4ded926d762c6fff8798 1.1.8

File: src/modules/email/email.service.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+export class EmailService {
+  constructor(
+    @Inject(forwardRef(() => UserService))
+    private userService: UserService,
+    @Inject(forwardRef(() => ConfigService))
+    private config: ConfigService,
+    private mailerService: MailerService,
+  ) {}
+
+  async sendEmailVerificationCode(email: string): Promise<any> {
```
Comment (lyudad) [inline src/modules/email/email.service.ts]: any это не тип

File: src/common/entities/user.entity.ts [hunk 4]
```diff
@@ -10,4 +11,18 @@ export class User {
...
+
+  @Column({
+    type: 'boolean',
+    name: 'is_verified',
+    default: false,
+  })
+  isVerified: boolean;
+
+  @Column({
+    type: 'varchar',
```
Comment (lyudad) [inline src/common/entities/user.entity.ts:23]: почему не int?

Comment (canterbery) [reply src/common/entities/user.entity.ts:23]: Потому что код может сгенерироваться с нулем на любой из 6 позиций.  Например 003402

**Pull request merged**